VoodooI2CHID/VoodooI2CTouchscreenHIDEventDriver.cpp

```diff
@@ -270,12 +280,16 @@ bool VoodooI2CTouchscreenHIDEventDriver::handleStart(IOService* provider) {
 
 void VoodooI2CTouchscreenHIDEventDriver::handleStop(IOService* provider) {
     if (timer_source) {
+        timer_source->cancelTimeout();
         work_loop->removeEventSource(timer_source);
         OSSafeReleaseNULL(timer_source);
     }
-    
-    OSSafeReleaseNULL(work_loop);
-    
+
+    if (work_loop) {
+        // OSSafeReleaseNULL(work_loop);
```

Comment:
You definetly want to release `work_loop`. In fact, `OSSafeReleaseNULL(work_loop);` is perfectly fine here.